pandas/core/arrays/categorical.py

```diff
@@ -1872,13 +1874,16 @@ def _repr_categories(self):
         )
         from pandas.io.formats import format as fmt
 
+        format_array = partial(
```

Comment:
since you changed this in pandas/io/format.py is it also necessary here?

Reply:
Yes, it's necessary so that `GenericArrayFormatter` is initialised with it